contracts/dice/dice.cpp

```diff
@@ -37,9 +35,10 @@ class dice : public eosio::contract {
       //@abi action
       void offerbet(const asset& bet, const account_name player, const checksum256& commitment) {
 
-         auto amount = eos_currency::token_type(bet);
+         eosio_assert( bet.symbol == S(4,EOS) , "only EOS token allowed" );
```

Comment:
Is this check sufficient to guarantee THE EOS token? Currently, I believe, multiple accounts can have an EOS token. Are we planning on having a symbol lib and guarantee only one token of each unique symbol?

Reply:
Yes, multiple accounts can create an _EOS_ token, but If **THE EOS** token will be hosted in the **eosio.token** contract (as I think it will be), the newest version of **eosio.token** can only host one token with _symbol_code == EOS_ (that hopefully will be **THE EOS** token).

Dice contract only interacts with **eosio.token**, so I think the check is sufficient.